app/next.config.js

```diff
@@ -40,15 +40,38 @@ module.exports = withPreconstruct(
       headers: async () => {
         const headers = [];
 
+        headers.push({
+          source: "/:path*",
+          headers: [
+            {
+              key: "X-Content-Type-Options",
+              value: "nosniff",
+            },
+          ],
+        });
+
+        // See https://content-security-policy.com/ & https://developers.google.com/tag-platform/security/guides/csp
+        if (!(process.env.DISABLE_CSP && process.env.DISABLE_CSP === "true")) {
+          headers[0].headers.push({
+            key: "Content-Security-Policy",
+            value: [
+              `default-src 'self' 'unsafe-inline'${process.env.NODE_ENV === "development" ? " 'unsafe-eval'" : ""} https://*.sentry.io https://vercel.live/ https://vercel.com https://*.googletagmanager.com`,
+              `script-src 'self' 'unsafe-inline'${process.env.NODE_ENV === "development" ? " 'unsafe-eval'" : ""} https://*.sentry.io https://vercel.live/ https://vercel.com https://*.googletagmanager.com`,
+              `style-src 'self' 'unsafe-inline'`,
+              `font-src 'self'`,
+              `form-action 'self'`,
+              `connect-src 'self' https://*.sentry.io https://*.vercel.app https://*.google-analytics.com https://*.analytics.google.com https://*.googletagmanager.com wss://*.pusher.com https://*.ldbar.ch`,
+              `img-src 'self' https://vercel.live https://vercel.com *.pusher.com *.pusherapp.com https://*.admin.ch https://*.opendataswiss.org https://*.google-analytics.com https://*.googletagmanager.com data: blob:`,
+              `script-src-elem 'self' 'unsafe-inline' https://*.admin.ch https://vercel.live https://vercel.com`,
+              `worker-src 'self' blob: https://*.admin.ch`,
+            ].join("; "),
+          });
+        }
+
         if (process.env.PREVENT_SEARCH_BOTS === "true") {
-          headers.push({
-            source: "/:path*",
-            headers: [
-              {
-                key: "X-Robots-Tag",
-                value: "noindex, nofollow",
-              },
-            ],
+          headers[0].headers.push({
+            key: "X-Robots-Tag",
+            value: "noindex, nofollow",
           });
         }
```

Comment:
@adintegra was this change to simplify the code, or needed to improve security? Just wondering out loud, as it looks that now we miss `"/:path*"` source :)